### DISCUSSION TITLE
Handle validation of multiple fields where only one is required

### DISCUSSION THREAD

**Discussion by FilipBrannlund** [react-hook-form/react-hook-form#5222] [Q&A] [answered]
What is the best practice to handle validation of multiple fields where only one is required?
In the example below I have an checkbox.
If the checkbox has been checked it shows 5 diffirent input fields.
One input field HAS to contain an value, it doesn't matter if it's field 1 or field 3.
https://codesandbox.io/s/mystifying-rgb-wvzly?file=/src/App.js

example 1
Checkbox has been checked
User does **not** fill in any field and pressed submit
Validation scans if any field contains value and an error message should pop up.

example 2
Checkbox has been checked
User fills in field 2 with an value
Validation sees field 2 has value and submit the field.

**Comment by leapful** [comment, marked as answer]
@FilipBrannlund 

You can use option "resolver" of "useForm" hook to implement a global conditional validation.

For example:
```
useForm({
   resolver: (values) => {
      if (values.otherReason) {
         // Do some validation on list reason
      }
      // Just remember to always return object { values, errors: {} } if valid, { values: {}, errors: { //some errors } } if invalid
   }
})
```

CSB sample: https://codesandbox.io/s/epic-sanne-yg46u?file=/src/App.js
